docker/bitcoind/start-bitcoind.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+
+# exit from script if error was raised.
+set -e
+
+# error function is used within a bash function in order to send the error
+# message directly to the stderr output and exit.
+error() {
+    echo "$1" > /dev/stderr
+    exit 0
+}
+
+# return is used within bash function in order to return the value.
+return() {
+    echo "$1"
+}
+
+# set_default function gives the ability to move the setting of default
+# env variable from docker file to the script thereby giving the ability to the
+# user override it during container start.
+set_default() {
+    # docker initialized env variables with blank string and we can't just
+    # use -z flag as usually.
+    BLANK_STRING='""'
+
+    VARIABLE="$1"
+    DEFAULT="$2"
+
+    if [[ -z "$VARIABLE" || "$VARIABLE" == "$BLANK_STRING" ]]; then
+
+        if [ -z "$DEFAULT" ]; then
+            error "You should specify default variable"
+        else
+            VARIABLE="$DEFAULT"
+        fi
+    fi
+
+   return "$VARIABLE"
+}
+
+# Set default variables if needed.
+RPCUSER=$(set_default "$RPCUSER" "devuser")
+RPCPASS=$(set_default "$RPCPASS" "devpass")
+NETWORK=$(set_default "$NETWORK" "regtest")
+DEBUG=$(set_default "$BITCOIND_DEBUG" "1")
+
+PARAMS=""
+if [ "$NETWORK" != "mainnet" ]; then
+   PARAMS="-$NETWORK"
+fi
+
+PARAMS=$(echo $PARAMS \
```

Comment:
We should probably add a section to the `README.md` that mentions how to use the `bitcoind` version of the compose file.
For example for mining, you need to use `bitcoin-cli -chain=regtest -rpcuser=devuser -rpcpassword=devpass generatetoaddress 10 bcrt1qyl948mkr0x37rjpzj59d828qkjlwpyzjz9tgm5` (or any regtest address).
And some other commands will probably be slightly different as well.